### PR TITLE
Initial arm64

### PR DESCRIPTION
**Pull request by manbearian**
# Initial ARM64 Work

Right now this is more of a proof-of-concept then anything else for #2346.

With these changes, after manually running the vcpkg_install script to install ARM64 packages, building the git.sln for ARM64 "just works", though much work remains before it is fully functional. The good news is that when i copied the built binaries onto my ARM64 device i can see some basic functionality working (i.e. the logo and help message). I then dropped the binary into a folder layout matching an installed version and things got a bit better: git checkout works (as long as no new files are needed), git init works, git --help and git --version work, and a few other things as well. Most things (e.g. checkout a file) do not work yet.

Summary of Changes
- project/sln changes (through perl script)
  - upgrade project to more recent VS (required to get ARM64 compilers, i picked latest, v142).
  - add ARM64 build configurations to projects/solution
  - added an item to the vcxprojs for VCPKGArch to be set correctly for ARM64
- source changes
  - update compat/bswap.h
- vcpkg changes
  - update compat/vcbuild/vcpg_install.bat to install both x64 and ARM64 packages

### CODE REVIEW

File: contrib/buildsystems/Generators/Vcxproj.pm [hunk 1]
```diff
@@ -139,7 +148,7 @@ sub createProject {
   </PropertyGroup>
   <PropertyGroup>
     <ConfigurationType>$config_type</ConfigurationType>
-    <PlatformToolset>v140</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
```
Comment (dscho) [inline contrib/buildsystems/Generators/Vcxproj.pm]: Could we keep `v140` here to keep things working with VS 2015? Upon loading the solution in Visual Studio, the user is asked whether they want to retarget anyway, and even MSBuild can override this via `/p:PlatformToolset=v142`.

**Comment by dennisameling** [comment]
@manbearian Do you need help with this? I'm also happy to take your PR as a base and build further on that (while keeping in your original commits, of course). Curious to see what's needed to get a minimal build that can be used e.g. by [`dugite-native`](https://github.com/desktop/dugite-native/blob/master/appveyor.yml).

**Comment by manbearian** [comment]
@dennisameling , thanks for offering to take a look. Please see if you can salvage my work on this. My priorities got spun around this year and its unlikely i will get an opportunity to pick this up again. I'm happy to help with any specific questions you might have on the work i did.

**Comment by dscho** [comment]
Shall we close this PR in favor of #2915?

**Pull request closed without merging**